doc/hacking/snmp-tunnels.rst

```diff
@@ -0,0 +1,85 @@
+=========================================
+Establishing SNMP tunnels using socat/SSH
+=========================================
+
+While developing new functionality related to SNMP collection in NAV,
+a developer's workstation isn't necessarily attached to a network that
+provides the developer with the necessary access to communicate with the
+require set of switches/routers etc. For internal equipment, this is
+usually solved by connection a VPN, but sometimes, the devices one needs
+to talk to are located at a customer site where there is no VPN access.
+
+In the case of having SSH access to a Linux server at the customer premises,
+that is allowed to communicate with the customer's switches and routers,
+one can establish a SNMP tunnel to the customer's equipment using a
+combination of SSH and socat.
+
+This guide will document a way to establish such a SNMP tunnel.
+
+
+
+Course of action
+================
+
+1. Forward a TCP port from your machine to another, which has the necessary
+   SNMP access level.
+
+   .. code-block:: sh
+
+      ssh -L 1161:localhost:1161 user@the-machine-with-access
+
+2. Then, on that port, set up a UDP-to-TCP tunnel using socat on that machine:
+
+   .. code-block:: sh
+
+       socat tcp4-listen:1161,reuseaddr,fork UDP:ip-or-name-of-switch-to-talk-to:161
+
+3. In a different terminal window, on our your localhost, set up a socat
+   tunnel to tunnel UDP traffic on port 161 through the forwarded TCP port
+   (sudo is necessary because you need to bind to port 161, which only root can do):
+
+   .. code-block:: sh
+
+      sudo socat -T15 udp4-recvfrom:161,reuseaddr,fork tcp:localhost:1161
+
+4. Finally to test connectivity in another terminal window:
+
+   .. code-block:: sh
+
+      snmpwalk -v2c -c public localhost SNMPv2-MIB::system
+
+   (replace public with the respective community string if it differs)
+
+5. When tunneling works you can add a new management profile with the
+   respective community string the switch uses to NAV.
+
+6. Then add an IP device with that management profile and your own IP
+   address, which you can find out by running:
+
+   .. code-block:: sh
+
+      curl ifconfig.me
+
+Troubleshooting
+===============
+
+- if in step 2 the error "Address already in use" appears, you can figure
+  out which process is using it by running
+
+  .. code-block:: sh
+
+      sudo netstat -aupn
+
+  and then potentially killing the process by running
+
+  .. code-block:: sh
+
+      sudo kill process_id
+
+  If the process restarts on its own it might help to run
```

Comment:
```suggestion
  If the process restarts on its own it might be that you need to kill its 
  parent process. This command can help identify the parent process:
```